### PR TITLE
refactor: implement `useCustomEvent` hook

### PR DESCRIPTION
**Pull request by adhrinae**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

- This PR makes the event bus easier to manage the lifecycle of custom event registration.

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: apps/calendar/src/hooks/common/__tests__/customEvent.spec.tsx [hunk 1]
```diff
@@ -17,18 +20,9 @@ describe('Event Bus Context', () => {
...
-      eb[fireOnce ? 'once' : 'on']('test', handler);
-    }, [eb, fireOnce, handler]);
-
-    return (
-      <div>
-        <button onClick={() => eb.fire('test')}>Fire Event</button>
-        <button onClick={() => eb.off('test')}>Remove Event</button>
-      </div>
-    );
+    const fire = useCustomEvent(TEST_EVENT_NAME, handler, fireOnce);
```
Comment (lja1018) [inline apps/calendar/src/hooks/common/__tests__/customEvent.spec.tsx:23]: off는 없어도 될까요??

**Review by lja1018 — APPROVED** [review]
리뷰 완료합니다.

File: apps/calendar/src/hooks/common/__tests__/customEvent.spec.tsx (hunk 1 above)
Comment (adhrinae) [reply apps/calendar/src/hooks/common/__tests__/customEvent.spec.tsx:23]: 처음에 고민은 했는데 직접 `off` 를 호출할 일이 없겠더라고요.
기본적으로 인스턴스의 destroy 호출할 때 이벤트버스의 모든 핸들러는 제거할거고, 더불어 이 훅을 쓰는 컴포넌트가 언마운트 될 때도 자동으로 해제될테니까요.

**Review by jajugoguma — APPROVED** [review]
리뷰 완료합니다.

**Pull request merged**